test/e2e/helloworld_shell_test.go

```diff
@@ -96,47 +96,50 @@ func TestHelloWorldFromShell(t *testing.T) {
 
 	logger.Info("Waiting for ingress to come up")
 
-	// Wait for ingress to come up
-	serviceIP := ""
-	serviceHost := ""
-	timeout := ingressTimeout
-	for (serviceIP == "" || serviceHost == "") && timeout >= 0 {
-		serviceHost = noStderrShell("kubectl", "get", "rt", "route-example", "-o", "jsonpath={.status.domain}", "-n", test.ServingNamespace)
-		serviceIP = noStderrShell("kubectl", "get", "svc", "istio-ingressgateway", "-n", "istio-system",
-			"-o", "jsonpath={.status.loadBalancer.ingress[*]['ip']}")
-		time.Sleep(checkInterval)
-		timeout = timeout - checkInterval
-	}
-	if serviceIP == "" || serviceHost == "" {
-		// serviceHost or serviceIP might contain a useful error, dump them.
-		t.Fatalf("Ingress not found (IP='%s', host='%s')", serviceIP, serviceHost)
-	}
-	logger.Infof("Ingress is at %s/%s", serviceIP, serviceHost)
-
-	logger.Info("Accessing app using curl")
-
-	outputString := ""
-	timeout = servingTimeout
-	for outputString != helloWorldExpectedOutput && timeout >= 0 {
-		var cmd *exec.Cmd
-		if test.ServingFlags.ResolvableDomain {
-			cmd = exec.Command("curl", serviceHost)
-		} else {
-			cmd = exec.Command("curl", "--header", "Host:"+serviceHost, "http://"+serviceIP)
+	gateways := []string{"istio-ingressgateway", "knative-ingressgateway"}
+	for _, gateway := range gateways {
+		// Wait for ingress to come up
+		serviceIP := ""
+		serviceHost := ""
+		timeout := ingressTimeout
+		for (serviceIP == "" || serviceHost == "") && timeout >= 0 {
+			serviceHost = noStderrShell("kubectl", "get", "rt", "route-example", "-o", "jsonpath={.status.domain}", "-n", test.ServingNamespace)
+			serviceIP = noStderrShell("kubectl", "get", "svc", gateway, "-n", "istio-system",
+				"-o", "jsonpath={.status.loadBalancer.ingress[*]['ip']}")
+			time.Sleep(checkInterval)
+			timeout = timeout - checkInterval
 		}
-		output, err := cmd.Output()
-		errorString := "none"
-		time.Sleep(checkInterval)
-		timeout = timeout - checkInterval
-		if err != nil {
-			errorString = err.Error()
+		if serviceIP == "" || serviceHost == "" {
+			// serviceHost or serviceIP might contain a useful error, dump them.
+			t.Fatalf("Ingress not found (IP='%s', host='%s')", serviceIP, serviceHost)
+		}
+		logger.Infof("Ingress is at %s/%s", serviceIP, serviceHost)
```

Comment:
These two can be joined:
Accessing app using curl at IP=%s DN=%s (or using curl to access app at...)

Reply:
Done